pkg/basicstation/cups/update_info.go

```diff
@@ -229,68 +235,77 @@ func (s *Server) UpdateInfo(c echo.Context) error {
 		gtw.Attributes = make(map[string]string)
 	}
 
-	if gtw.LBSLNSSecret == nil {
-		return errLNSCredentials.WithAttributes("gateway_uid", uid)
-	}
-
-	if s.requireExplicitEnable || gtw.Attributes[cupsAttribute] != "" {
+	if s.requireExplicitEnable && gtw.Attributes[cupsAttribute] != "" {
 		if cups, _ := strconv.ParseBool(gtw.Attributes[cupsAttribute]); !cups {
 			return errCUPSNotEnabled.WithAttributes("gateway_uid", uid)
 		}
 	}
 
 	res := UpdateInfoResponse{}
 
-	if gtw.Attributes[cupsURIAttribute] == "" {
-		gtw.Attributes[cupsURIAttribute] = req.CUPSURI
-	}
-	if s.allowCUPSURIUpdate && gtw.Attributes[cupsURIAttribute] != req.CUPSURI {
-		res.CUPSURI = gtw.Attributes[cupsURIAttribute]
+	_, _, _, err = parseAddress("https", req.CUPSURI)
+	if err != nil {
+		return errInvalidCUPSURI.WithAttributes("uri", req.CUPSURI)
 	}
```

Comment:
It's technically possible for it to make a request to a valid URI but set a nonsense value to the `cupsURI` field in the request body. But this would fail in fetching the trust to I've improved the error logging down the line.